### ISSUE TITLE
Lambda Runtime out of support support warnings

### ISSUE THREAD

**Issue by GeorgeBellTMH** [aws-amplify/amplify-cli#11007]
### Describe the feature you'd like to request

I'd like to propose a suite of features that will help relieve some of the risk associated with old environments running end-of-life runtimes (where a rollback becomes impossible without manual edits of the lambda in AWS).

1. All built in amplify lambda's (both generated and stock) should default to the most recent version of the lambda runtime when created. This should be obvious.
2. If the version of amplify is so old that we are creating lambdas that won't be able to be deployed we should warn...
3. All updates to lambda's that aren't on the most recent runtime should throw a warning recommending upgrading to the most recent version (and an automated way of doing that)
4. All push's where there is an existing lambda that is running an out of support runtime should throw a very big warning (regarding rollback) and ask to continue - recommending that the lambda be updated without other changes that may force a rollback
5. When switching to an environment, we should check all lambda's to ensure they are on latest version, and throw warnings

### Describe the solution you'd like

Basically we need a way to check and validate that functions will deploy and fallback successfully before we actually do the deployment.

### Describe alternatives you've considered

Nothing

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change

**Comment by josefaidt** [comment]
Hey @GeorgeBellTMH :wave: thanks for raising this! I think these are all very valid points and a great step in the right direction. As such, I've marked this as a feature request for the team to evaluate further 🙂
